shop/src/main/java/ro/msg/learning/shop/service/StockService.java

```diff
@@ -0,0 +1,137 @@
+package ro.msg.learning.shop.service;
+
+import jakarta.transaction.Transactional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.msg.learning.shop.domain.Location;
+import ro.msg.learning.shop.domain.Product;
+import ro.msg.learning.shop.domain.Stock;
+import ro.msg.learning.shop.domain.StockId;
+import ro.msg.learning.shop.dto.ProductIdAndQuantityDto;
+import ro.msg.learning.shop.dto.StockCreateDto;
+import ro.msg.learning.shop.dto.StockGetDto;
+import ro.msg.learning.shop.exception.ResourceNotFoundException;
+import ro.msg.learning.shop.exception.StockUpdateException;
+import ro.msg.learning.shop.mapper.StockMapper;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.ProductRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import java.util.List;
+import java.util.UUID;
+
+@Service
+public class StockService {
+    public static final String STOCK_WITH_ID = "Stock with ID '";
+    public static final String NOT_FOUND = "' not found.";
+    public static final String PRODUCT_WITH_ID = "Product with ID '";
+    public static final String LOCATION_WITH_ID = "Location with ID '";
+    public static final String PRODUCT_NOT_FOUND_WITH_ID = "Product not found with ID:";
+    public static final String PRODUCT_WITH_ID1 = "Stock not found for product with ID: ";
+    public static final String INSUFFICIENT_STOCK = "Insufficient stock";
+    @Autowired
+    private StockRepository stockRepository;
+    @Autowired
+    private ProductRepository productRepository;
+    @Autowired
+    private LocationRepository locationRepository;
```

Comment:
the methods which you need to access from these repositories are available in each service. Help services communicate with each other but let each service communicate only with its concerning repos

Reply:
Solved.